src/memoization/memoization.jl

```diff
@@ -0,0 +1,48 @@
+mutable struct MemoNLP <: AbstractNLPModel
+  model :: AbstractNLPModel
+  meta :: NLPModelMeta
+  counters :: Counters
+  min_elm :: Int
+  max_elm :: Int
+  memo :: Dict{AbstractVector, AbstractFloat}
+  freq :: Dict{AbstractVector, Int}
+end
+
+function MemoNLP(model :: AbstractNLPModel, max_elm :: Int, min_elm :: Int)
+  max_elm != 0 || return model
```

Comment:
I'd prefer to use a boolean flag instead of an integer, because i think it's more "user-friendly". But since that's definitely not top priority, i'd say this does the trick. Let me know what you think !